packages/types/src/payment-types.ts

```diff
@@ -118,11 +144,11 @@ export interface IPaymentNetworkEvent<TEventParameters, TEventNames = EVENTS_NAM
  */
 
 /** Parameters for events of Declarative payments */
-export interface IDeclarativePaymentEventParameters {
+export interface IDeclarativePaymentEventParameters<TFrom = IIdentity> {
```

Comment:
(cf. other comment about this interface) Declarative payment events are at the root of many detectors, but when events are declarative for real, they use IIdentity type for from, whereas it's string for overriding children (when it's not declarative)